### ISSUE TITLE
Forward slash '/' cannot be used in object names

### ISSUE THREAD

**Issue by cbachert** [fortinet-solutions-cse/40ansible#48]
The character forward slash '/' cannot be used in object names, for example:
```
address_subnet:
  - name: 'test-10.0.0.1/32'
    subnet: '10.0.0.1 255.255.255.255'
```

When using the literal '/' this is regarded as an URL separator which makes the API PUT request fail with HTTP 400. See also httpsd debug log and packet capture:
[slash-literal.txt](https://github.com/fortinet-solutions-cse/40ansible/files/3896619/slash-literal.txt)
[slash-literal.zip](https://github.com/fortinet-solutions-cse/40ansible/files/3896620/slash-literal.zip)

When encoding the '/' as '%2F' the API PUT and POST requests actually go through, but the object is created with '%2F' in its name instead of '/'. See also httpsd debug log and packet capture:
[slash-encoded.txt](https://github.com/fortinet-solutions-cse/40ansible/files/3896617/slash-encoded.txt)
[slash-encoded.zip](https://github.com/fortinet-solutions-cse/40ansible/files/3896618/slash-encoded.zip)

Potentially this may be a bug at fortiosapi-level, I may raise there if you can confirm that this lies with that project.

**Comment by frankshen01** [comment]
Hi @cbachert We reproduced the bug you submitted and will discuss the solution with the fortiapi developers. Thank you!

**Comment by chillancezen** [comment]
This looks like a duplication of issue: https://github.com/fortinet-solutions-cse/ansible_fgt_modules/issues/14 
 
I made a fix from ansible side with which you don't have to encode your name yourself, ansible will do it for you. https://github.com/ansible/ansible/pull/66050


from the log slash-encoded.txt, `test-10.0.0.1%2F32` is resolved back to `test-10.0.0.1/32` correctly. @cbachert can you help verify the fix with latest fortigate build? if this issue persists, could you share your build number and ansible version?

thanks,
Link

**Comment by chillancezen** [comment]
> This looks like a duplication of issue: [fortinet-solutions-cse/ansible_fgt_modules#14](https://github.com/fortinet-solutions-cse/ansible_fgt_modules/issues/14)
> 
> I made a fix from ansible side with which you don't have to encode your name yourself, ansible will do it for you. [ansible/ansible#66050](https://github.com/ansible/ansible/pull/66050)
> 
> from the log slash-encoded.txt, `test-10.0.0.1%2F32` is resolved back to `test-10.0.0.1/32` correctly. @cbachert can you help verify the fix with latest fortigate build? if this issue persists, could you share your build number and ansible version?
> 
> thanks,
> Link

I realized this project call fortiosapi library, there is no url encoding for mkey in fortiosapi now, the fix in last comment is httpapi only. 

the fix is not for this.

**Comment by JieX19** [comment]
Hi @cbachert, we've fixed the issue and you will be able to use "/" in the object names shortly. Thanks for your question!
